src/Libraries/RevitNodes/Elements/Tag.cs

```diff
@@ -0,0 +1,269 @@
+            Autodesk.Revit.DB.Element host,
+            Autodesk.Revit.DB.TagOrientation orientation,
+            Autodesk.Revit.DB.TagMode mode,
+            bool addLeader,
+            Autodesk.Revit.DB.XYZ point)
+        {
+            SafeInit(() => Init(view, host, orientation, mode, addLeader, point));
+        }
+
+        #endregion
+
+        #region Helpers for private constructors
+
+        /// <summary>
+        /// Set Text
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="location"></param>
+        /// <param name="angle"></param>
+        private void InternalSetType(string value, XYZ location, double angle)
+        {
+            string font = "Arial";
+            double size = 0.0104;
+            InternalSetTextSettings(value, false, false, size, font, location.ToPoint(true), HorizontalTextAlignment.Center, angle);
+        }
+
+        /// <summary>
+        /// Init element from existing
+        /// </summary>
+        /// <param name="element"></param>
+        private void InitElement(Autodesk.Revit.DB.IndependentTag element)
+        {
+            double rotation = (element.TagOrientation == TagOrientation.Horizontal) ? 0 : 90;
+            InternalSetType(element.TagText, element.TagHeadPosition, rotation);
+            InternalSetElement(element);
+        }
+
+        /// <summary>
+        /// Init Tag element
+        /// </summary>
+        /// <param name="view"></param>
+        /// <param name="host"></param>
+        /// <param name="orientation"></param>
+        /// <param name="mode"></param>
+        /// <param name="addLeader"></param>
+        /// <param name="point"></param>
+        private void Init(
+            Autodesk.Revit.DB.View view,
+            Autodesk.Revit.DB.Element host,
+            Autodesk.Revit.DB.TagOrientation orientation,
+            Autodesk.Revit.DB.TagMode mode,
+            bool addLeader,
+            Autodesk.Revit.DB.XYZ point)
+        {
+
+            Autodesk.Revit.DB.Document document = DocumentManager.Instance.CurrentDBDocument;
+            TransactionManager.Instance.EnsureInTransaction(document);
+
+            var tagElem = ElementBinder.GetElementFromTrace<Autodesk.Revit.DB.IndependentTag>(document);
+
+            // create a new tag if the existing tag is null or its host view is not the selected one
+            // or the host is neither set a host or linked element.
+            if (tagElem == null || view.Id != tagElem.OwnerViewId ||
+                (tagElem.TaggedElementId.HostElementId != host.Id && tagElem.TaggedElementId.LinkedElementId != host.Id))
+            {
+                tagElem = document.Create.NewTag(view, host, addLeader, mode, orientation, point);
+            }
+            else
+            {
+                // apply properties
+                tagElem.TagOrientation = orientation;
+                tagElem.HasLeader = addLeader;
+                tagElem.TagHeadPosition = point;
```

Comment:
does this work? you may want to check these properties and not set them again if they are the same values as those already on the element to avoid infinite loops.